client/plots/brainImaging.js

```diff
@@ -186,9 +205,19 @@ class BrainImaging {
 			config.overlayTW?.term?.id !== previousConfig?.overlayTW?.term?.id ||
 			config.divideByTW?.term?.id !== previousConfig?.divideByTW?.term?.id
 
-		const updateL = config.settings.brainImaging.brainImageL != this.settings?.brainImageL || overlayOrDivideByChange
-		const updateF = config.settings.brainImaging.brainImageF != this.settings?.brainImageF || overlayOrDivideByChange
-		const updateT = config.settings.brainImaging.brainImageT != this.settings?.brainImageT || overlayOrDivideByChange
+		const legendFilterChange = JSON.stringify(previousConfig?.legendFilter) !== JSON.stringify(config.legendFilter)
```

Comment:
I wonder why this is needed, seems that with the other variables: updateL, updateF, updateT and overlayOrDivideByChange all the changes in the state can be tracked.

Reply:
legendFilterChange is for tracking the legendFilter change. e.g.Hide Male, then we need to re-render images and legends